spi/src/main/java/org/opensearch/security/spi/resources/ResourceProvider.java

```diff
@@ -20,4 +20,13 @@ public interface ResourceProvider {
 
     String resourceIndexName();
 
+    /**
+     * Returns the name of the field representing the resource type in the resource document.
+     *
+     * @return the field name containing the resource type
+     */
+    default String typeField() {
```

Comment:
we should make this mandatory. However this will also require changes on plugin side to now store the type in their resource document. OR am i reading this wrong and having that in-memory would work?